beacon_node/store/src/database/redb_impl.rs

```diff
@@ -0,0 +1,323 @@
+use crate::{metrics, ColumnIter, ColumnKeyIter, Key};
+use crate::{DBColumn, Error, KeyValueStoreOp};
+use parking_lot::{Mutex, MutexGuard, RwLock};
+use redb::TableDefinition;
+use std::collections::HashSet;
+use std::{borrow::BorrowMut, marker::PhantomData, path::Path};
+use strum::IntoEnumIterator;
+use types::EthSpec;
+
+use super::interface::WriteOptions;
+
+pub const DB_FILE_NAME: &str = "database.redb";
+
+pub struct Redb<E: EthSpec> {
+    db: RwLock<redb::Database>,
+    transaction_mutex: Mutex<()>,
+    _phantom: PhantomData<E>,
+}
+
+impl From<WriteOptions> for redb::Durability {
+    fn from(options: WriteOptions) -> Self {
+        if options.sync {
+            redb::Durability::Immediate
+        } else {
+            redb::Durability::Eventual
+        }
+    }
+}
+
+impl<E: EthSpec> Redb<E> {
+    pub fn open(path: &Path) -> Result<Self, Error> {
+        let path = if path.is_dir() {
+            path.join(DB_FILE_NAME)
+        } else {
+            path.to_path_buf()
+        };
```

Comment:
This looks error-prone to me. If the path is intended as a directory that doesn't exist yet (like `$datadir/beacon/chain_db`), it seems strange to use it as a file path. I think we would be better off always appending `DB_FILE_NAME`.

Reply:
Removed this here https://github.com/sigp/lighthouse/pull/4718/commits/1feac02af96e7120a14a7366a5bc829d33f4faf6